### ISSUE TITLE
ci: nodejs warnings

### ISSUE THREAD

**Issue by williballenthin** [williballenthin/lancelot#172]
```
Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: actions/checkout@v2, actions/upload-artifact@v2
```

here: https://github.com/williballenthin/lancelot/actions/runs/3685473688

**Comment by williballenthin** [comment]
i think we can upgrade to `upload-artifact/v3` without issue: 
https://github.com/actions/upload-artifact/releases/tag/v3.0.0

**Comment by williballenthin** [comment]
`checkout/v3` should be ok too:
https://github.com/actions/checkout/releases/tag/v3.0.0

**Comment by williballenthin** [comment]
closed in afb0b75e86e29addb65146f3ab312803d18ea177
